### ISSUE TITLE
addField() deprecation warning

### ISSUE THREAD

**Issue by wonder-sk** [hobu/laz-perf#122]
The commit https://github.com/hobu/laz-perf/commit/e0bd817dd15cf57aa04d8b6b3ae809fbc1af84bc marks `eb_vlr::addField()` as deprecated, but it's still used in the code. When I use the code in QGIS, I get the following warning (which is a problem because in CI warnings are treated as errors):

```
[5/24] Building CXX object src/core/CMakeFiles/qgis_core.dir/__/__/external/lazperf/vlr.cpp.o
../external/lazperf/vlr.cpp: In constructor ‘lazperf::eb_vlr::eb_vlr(int)’:
../external/lazperf/vlr.cpp:329:18: warning: ‘void lazperf::eb_vlr::addField()’ is deprecated [-Wdeprecated-declarations]
  329 |         addField();
      |                  ^
In file included from ../external/lazperf/vlr.cpp:37:
../external/lazperf/vlr.hpp:166:25: note: declared here
  166 |     [[deprecated]] void addField();
      |                         ^~~~~~~~
```

**Comment by abellgithub** [comment]
The function in which it is used is also marked as deprecated.

**Comment by wonder-sk** [comment]
The other function does not get used, so it is not flagged by the compiler...

Is it feasible to maybe remove both of them to avoid the warning?

**Comment by wonder-sk** [comment]
I believe it is GCC emitting these warnings, with `-Wdeprecated-declarations` flag enabled (not sure if clang supports this flag as well)

**Comment by abellgithub** [comment]
I guess I'm not sure what you're doing. Why are you changing the build flags when you compile lazperf? It makes sense that you would set "-Wdeprecated-declarations" when building QGIS, but we set "-Wno-deprecated-declarations" on purpose for lazperf. I guess we could surround the code in that function with a pragma to turn it off, but it seems like a general issue for you if you're going to change the build flags that this package is expecting.

**Comment by wonder-sk** [comment]
This is my situation:
1. I have embedded the new version of lazperf into QGIS source code (https://github.com/qgis/QGIS/pull/47747)
2. QGIS code goes through a bunch of github workflows that build QGIS on various platforms and run tests. At least on GCC builds, the `-Wdeprecated-declarations` flag is set as enabled + the flag `-Werror` is used. So the build on these platforms was failing due to deprecation warnings turned into errors.
3. I have temporarily fixed it by patching lazperf code to remove the deprecation flag, but I am not a big fan of keeping extra patches

So the idea of this ticket was that if I can get rid of the deprecation warnings in the code, things will be easier as I don't need to patch lazperf. Now I realize I could selectively disable deprecation warnings just for lazperf code - if your preference is to keep those bits of code in lazperf deprecated, I guess I will just do the updates to QGIS build system to skip these warnings for lazperf code...

**Comment by abellgithub** [comment]
I'm sorry. I didn't realize that we were calling `eb_vlr::eb_vlr(int)` from a non-deprecated function. I'll remove its use.

**Comment by wonder-sk** [comment]
One more deprecation warning - this time `eb_vlr::eb_vlr(int)` being called from writers.cpp:
```
../external/lazperf/writers.cpp: In member function ‘void lazperf::writer::basic_file::Private::writeHeader()’:
../external/lazperf/writers.cpp:199:34: warning: ‘lazperf::eb_vlr::eb_vlr(int)’ is deprecated [-Wdeprecated-declarations]
  199 |     eb_vlr ebVlr(head14.ebCount());
      |                                  ^
```

**Comment by abellgithub** [comment]
Fixed with #128
